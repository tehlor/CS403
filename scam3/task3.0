(define dq (deque))
((dq'enqueueFront) 3)
((dq'enqueueBack) 5)
((dq'enqueueIndex) 1 4)
((dq'display))
(println)
