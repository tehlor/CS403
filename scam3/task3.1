(define dq (deque))
((dq'enqueueIndex) 0 41)
((dq'enqueueIndex) 1 99)
((dq'enqueueIndex) 1 93)
((dq'enqueueIndex) 1 18)
((dq'enqueueIndex) 2 56)
((dq'enqueueIndex) 4 81)
((dq'enqueueIndex) 2 83)
((dq'enqueueIndex) 1 99)
((dq'enqueueIndex) 4 51)
((dq'enqueueIndex) 3 33)
((dq'display))
(println)
