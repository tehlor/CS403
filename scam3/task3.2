(define dq (deque))
((dq'enqueueIndex) 0 53)
((dq'enqueueIndex) 0 73)
((dq'enqueueIndex) 2 22)
((dq'enqueueIndex) 0 59)
((dq'enqueueIndex) 0 88)
((dq'enqueueIndex) 5 92)
((dq'enqueueIndex) 2 86)
((dq'enqueueIndex) 0 19)
((dq'enqueueIndex) 2 20)
((dq'enqueueIndex) 8 90)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
