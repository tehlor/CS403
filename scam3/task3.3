(define dq (deque))
((dq'enqueueIndex) 0 20)
((dq'enqueueIndex) 0 46)
((dq'enqueueIndex) 0 40)
((dq'enqueueIndex) 2 54)
((dq'enqueueIndex) 2 74)
((dq'enqueueIndex) 5 12)
((dq'enqueueIndex) 6 87)
((dq'enqueueIndex) 3 27)
((dq'enqueueIndex) 5 32)
((dq'enqueueIndex) 7 55)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'display))
(println)
((dq'enqueueIndex) 0 68)
((dq'enqueueIndex) 1 85)
((dq'enqueueIndex) 0 88)
((dq'enqueueIndex) 0 77)
((dq'enqueueIndex) 1 78)
((dq'enqueueIndex) 2 42)
((dq'enqueueIndex) 5 45)
((dq'enqueueIndex) 1 35)
((dq'enqueueIndex) 5 85)
((dq'enqueueIndex) 5 43)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
