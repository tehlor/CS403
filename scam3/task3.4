(define dq (deque))
((dq'enqueueIndex) 0 50)
((dq'enqueueIndex) 0 65)
((dq'enqueueIndex) 0 36)
((dq'enqueueIndex) 1 66)
((dq'enqueueIndex) 2 18)
((dq'enqueueIndex) 3 95)
((dq'enqueueIndex) 2 91)
((dq'enqueueIndex) 2 94)
((dq'enqueueIndex) 8 72)
((dq'enqueueIndex) 1 74)
((dq'enqueueIndex) 5 83)
((dq'enqueueIndex) 5 17)
((dq'enqueueIndex) 7 78)
((dq'enqueueIndex) 11 56)
((dq'enqueueIndex) 9 60)
((dq'enqueueIndex) 10 41)
((dq'enqueueIndex) 0 77)
((dq'enqueueIndex) 17 15)
((dq'enqueueIndex) 0 47)
((dq'enqueueIndex) 13 64)
((dq'enqueueIndex) 10 40)
((dq'enqueueIndex) 12 89)
((dq'enqueueIndex) 5 85)
((dq'enqueueIndex) 19 21)
((dq'enqueueIndex) 13 95)
((dq'enqueueIndex) 21 15)
((dq'enqueueIndex) 20 33)
((dq'enqueueIndex) 3 39)
((dq'enqueueIndex) 0 96)
((dq'enqueueIndex) 25 67)
((dq'enqueueIndex) 15 53)
((dq'enqueueIndex) 31 56)
((dq'enqueueIndex) 7 96)
((dq'enqueueIndex) 19 35)
((dq'enqueueIndex) 13 33)
((dq'enqueueIndex) 31 90)
((dq'enqueueIndex) 22 49)
((dq'enqueueIndex) 29 85)
((dq'enqueueIndex) 10 63)
((dq'enqueueIndex) 38 83)
((dq'enqueueIndex) 22 83)
((dq'enqueueIndex) 36 37)
((dq'enqueueIndex) 3 10)
((dq'enqueueIndex) 27 19)
((dq'enqueueIndex) 43 52)
((dq'enqueueIndex) 34 52)
((dq'enqueueIndex) 44 75)
((dq'enqueueIndex) 47 26)
((dq'enqueueIndex) 33 61)
((dq'enqueueIndex) 23 16)
((dq'enqueueIndex) 42 42)
((dq'enqueueIndex) 50 49)
((dq'enqueueIndex) 42 76)
((dq'enqueueIndex) 14 16)
((dq'enqueueIndex) 18 32)
((dq'enqueueIndex) 49 88)
((dq'enqueueIndex) 3 78)
((dq'enqueueIndex) 10 22)
((dq'enqueueIndex) 45 83)
((dq'enqueueIndex) 14 76)
((dq'enqueueIndex) 17 98)
((dq'enqueueIndex) 13 32)
((dq'enqueueIndex) 44 28)
((dq'enqueueIndex) 27 46)
((dq'enqueueIndex) 50 91)
((dq'enqueueIndex) 31 65)
((dq'enqueueIndex) 17 49)
((dq'enqueueIndex) 4 14)
((dq'enqueueIndex) 12 40)
((dq'enqueueIndex) 8 55)
((dq'enqueueIndex) 41 11)
((dq'enqueueIndex) 27 67)
((dq'enqueueIndex) 56 61)
((dq'enqueueIndex) 57 58)
((dq'enqueueIndex) 29 12)
((dq'enqueueIndex) 20 71)
((dq'enqueueIndex) 0 53)
((dq'enqueueIndex) 72 74)
((dq'enqueueIndex) 55 42)
((dq'enqueueIndex) 9 53)
((dq'enqueueIndex) 21 63)
((dq'enqueueIndex) 8 57)
((dq'enqueueIndex) 2 24)
((dq'enqueueIndex) 48 28)
((dq'enqueueIndex) 42 73)
((dq'enqueueIndex) 61 17)
((dq'enqueueIndex) 62 19)
((dq'enqueueIndex) 63 53)
((dq'enqueueIndex) 60 55)
((dq'enqueueIndex) 2 16)
((dq'enqueueIndex) 48 37)
((dq'enqueueIndex) 69 58)
((dq'display))
(println)
((dq'dequeueIndex) 72)
((dq'display))
(println)
((dq'dequeueIndex) 62)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 43)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 83)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 80)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 68)
((dq'display))
(println)
((dq'dequeueIndex) 79)
((dq'display))
(println)
((dq'dequeueIndex) 18)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 68)
((dq'display))
(println)
((dq'dequeueIndex) 60)
((dq'display))
(println)
((dq'dequeueIndex) 60)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 38)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 47)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 23)
((dq'display))
(println)
((dq'dequeueIndex) 50)
((dq'display))
(println)
((dq'dequeueIndex) 37)
((dq'display))
(println)
((dq'dequeueIndex) 41)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 27)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 21)
((dq'display))
(println)
((dq'dequeueIndex) 54)
((dq'display))
(println)
((dq'dequeueIndex) 14)
((dq'display))
(println)
((dq'dequeueIndex) 40)
((dq'display))
(println)
((dq'dequeueIndex) 49)
((dq'display))
(println)
((dq'dequeueIndex) 30)
((dq'display))
(println)
((dq'dequeueIndex) 36)
((dq'display))
(println)
((dq'dequeueIndex) 50)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 33)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 39)
((dq'display))
(println)
((dq'dequeueIndex) 40)
((dq'display))
(println)
((dq'dequeueIndex) 43)
((dq'display))
(println)
((dq'dequeueIndex) 41)
((dq'display))
(println)
((dq'dequeueIndex) 33)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 23)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 27)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 9)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'display))
(println)
((dq'enqueueIndex) 0 84)
((dq'enqueueIndex) 0 32)
((dq'enqueueIndex) 0 52)
((dq'enqueueIndex) 1 70)
((dq'enqueueIndex) 1 25)
((dq'enqueueIndex) 0 46)
((dq'enqueueIndex) 4 60)
((dq'enqueueIndex) 3 98)
((dq'enqueueIndex) 0 37)
((dq'enqueueIndex) 0 36)
((dq'enqueueIndex) 10 98)
((dq'enqueueIndex) 10 77)
((dq'enqueueIndex) 4 75)
((dq'enqueueIndex) 13 93)
((dq'enqueueIndex) 8 32)
((dq'enqueueIndex) 4 45)
((dq'enqueueIndex) 1 58)
((dq'enqueueIndex) 11 43)
((dq'enqueueIndex) 0 11)
((dq'enqueueIndex) 0 32)
((dq'enqueueIndex) 3 15)
((dq'enqueueIndex) 14 85)
((dq'enqueueIndex) 14 14)
((dq'enqueueIndex) 19 71)
((dq'enqueueIndex) 8 92)
((dq'enqueueIndex) 25 40)
((dq'enqueueIndex) 24 84)
((dq'enqueueIndex) 2 31)
((dq'enqueueIndex) 16 14)
((dq'enqueueIndex) 5 18)
((dq'enqueueIndex) 9 49)
((dq'enqueueIndex) 15 44)
((dq'enqueueIndex) 32 21)
((dq'enqueueIndex) 25 98)
((dq'enqueueIndex) 4 82)
((dq'enqueueIndex) 8 39)
((dq'enqueueIndex) 32 91)
((dq'enqueueIndex) 6 54)
((dq'enqueueIndex) 36 98)
((dq'enqueueIndex) 6 37)
((dq'enqueueIndex) 37 23)
((dq'enqueueIndex) 27 82)
((dq'enqueueIndex) 42 76)
((dq'enqueueIndex) 2 57)
((dq'enqueueIndex) 35 29)
((dq'enqueueIndex) 28 19)
((dq'enqueueIndex) 30 20)
((dq'enqueueIndex) 23 66)
((dq'enqueueIndex) 11 32)
((dq'enqueueIndex) 31 43)
((dq'enqueueIndex) 3 87)
((dq'enqueueIndex) 36 93)
((dq'enqueueIndex) 40 88)
((dq'enqueueIndex) 22 73)
((dq'enqueueIndex) 47 63)
((dq'enqueueIndex) 0 78)
((dq'enqueueIndex) 42 69)
((dq'enqueueIndex) 32 75)
((dq'enqueueIndex) 23 65)
((dq'enqueueIndex) 15 27)
((dq'enqueueIndex) 50 89)
((dq'enqueueIndex) 18 53)
((dq'enqueueIndex) 62 80)
((dq'enqueueIndex) 7 31)
((dq'enqueueIndex) 2 76)
((dq'enqueueIndex) 40 18)
((dq'enqueueIndex) 40 38)
((dq'enqueueIndex) 1 42)
((dq'enqueueIndex) 12 50)
((dq'enqueueIndex) 5 13)
((dq'enqueueIndex) 2 17)
((dq'enqueueIndex) 57 81)
((dq'enqueueIndex) 54 42)
((dq'enqueueIndex) 38 22)
((dq'enqueueIndex) 73 80)
((dq'enqueueIndex) 25 82)
((dq'enqueueIndex) 51 67)
((dq'enqueueIndex) 23 69)
((dq'enqueueIndex) 33 47)
((dq'enqueueIndex) 71 51)
((dq'enqueueIndex) 12 55)
((dq'enqueueIndex) 45 78)
((dq'enqueueIndex) 68 62)
((dq'enqueueIndex) 10 10)
((dq'enqueueIndex) 1 27)
((dq'enqueueIndex) 3 15)
((dq'enqueueIndex) 24 86)
((dq'enqueueIndex) 75 12)
((dq'enqueueIndex) 18 43)
((dq'enqueueIndex) 15 26)
((dq'enqueueIndex) 14 55)
((dq'enqueueIndex) 91 83)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 42)
((dq'display))
(println)
((dq'dequeueIndex) 49)
((dq'display))
(println)
((dq'dequeueIndex) 62)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 73)
((dq'display))
(println)
((dq'dequeueIndex) 73)
((dq'display))
(println)
((dq'dequeueIndex) 47)
((dq'display))
(println)
((dq'dequeueIndex) 51)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 59)
((dq'display))
(println)
((dq'dequeueIndex) 54)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 72)
((dq'display))
(println)
((dq'dequeueIndex) 55)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 42)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 18)
((dq'display))
(println)
((dq'dequeueIndex) 54)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 28)
((dq'display))
(println)
((dq'dequeueIndex) 64)
((dq'display))
(println)
((dq'dequeueIndex) 40)
((dq'display))
(println)
((dq'dequeueIndex) 59)
((dq'display))
(println)
((dq'dequeueIndex) 61)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 53)
((dq'display))
(println)
((dq'dequeueIndex) 35)
((dq'display))
(println)
((dq'dequeueIndex) 56)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 34)
((dq'display))
(println)
((dq'dequeueIndex) 46)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 24)
((dq'display))
(println)
((dq'dequeueIndex) 42)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 25)
((dq'display))
(println)
((dq'dequeueIndex) 23)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 12)
((dq'display))
(println)
((dq'dequeueIndex) 32)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 35)
((dq'display))
(println)
((dq'dequeueIndex) 33)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 25)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 3)
((dq'display))
(println)
((dq'dequeueIndex) 22)
((dq'display))
(println)
((dq'dequeueIndex) 31)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 20)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 26)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 19)
((dq'display))
(println)
((dq'dequeueIndex) 23)
((dq'display))
(println)
((dq'dequeueIndex) 13)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 17)
((dq'display))
(println)
((dq'dequeueIndex) 15)
((dq'display))
(println)
((dq'dequeueIndex) 11)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 10)
((dq'display))
(println)
((dq'dequeueIndex) 16)
((dq'display))
(println)
((dq'dequeueIndex) 7)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 8)
((dq'display))
(println)
((dq'dequeueIndex) 5)
((dq'display))
(println)
((dq'dequeueIndex) 6)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 4)
((dq'display))
(println)
((dq'dequeueIndex) 2)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
((dq'dequeueIndex) 1)
((dq'display))
(println)
((dq'dequeueIndex) 0)
((dq'display))
(println)
