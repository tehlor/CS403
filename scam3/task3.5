(include "random.lib")
(randomSeed 76)
(define dq (deque))
(define (upfront dq n)
    (define value (randomRange 10 100))
    (cond
        ((= n 0)
            ((dq'enqueueFront) value)
            )
        ((< (dq'size) 10)
            ((dq'enqueueFront) value)
            (upfront dq (- n 1))
            )
        (else
            ((dq'enqueueIndex) (randomRange 2 10) value)
            (upfront dq (- n 1))
            )
        )
    )
(define (frontback dq n)
    (define (remove x mode)
        (cond
            ((= x n) (println "...]"))
            (else
                (print ((dq mode)) ",")
                (remove (+ x 1) mode)
                )
            )
        )
    ;((dq'display)) (println)
    (print "[")
    (remove 0 'dequeueFront)
    (print "[")
    (remove 0 'dequeueBack)
    )
(upfront dq 3037)
(frontback dq 20)
