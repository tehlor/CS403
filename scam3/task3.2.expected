[19,88,20,59,86,73,53,22,90,92]
[19,88,59,86,73,53,22,90,92]
[19,88,59,86,73,53,22,90]
[19,88,86,73,53,22,90]
[88,86,73,53,22,90]
[86,73,53,22,90]
[86,73,53,22]
[73,53,22]
[73,22]
[22]
[]
